src/System.Data.Common/src/System/Data/Common/AdapterUtil.cs

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading.Tasks;
+using Res = System.SR;
+
+namespace System
+{
+    internal static partial class SR
+    {
+        internal static string GetString(string value)
+        {
+            return value;
+        }
+
+        internal static string GetString(string format, params object[] args)
+        {
+            return SR.Format(format, args);
+        }
+    }
+}
+
+
+namespace System.Data.Common
+{
+    internal static class ADP
+    {
+        // The class ADP defines the exceptions that are specific to the Adapters.f
+        // The class contains functions that take the proper informational variables and then construct
+        // the appropriate exception with an error string obtained from the resource Framework.txt.
+        // The exception is then returned to the caller, so that the caller may then throw from its
+        // location so that the catcher of the exception will have the appropriate call stack.
+        // This class is used so that there will be compile time checking of error messages.
+        // The resource Framework.txt will ensure proper string text based on the appropriate
+        // locale.
+
+        static internal Task<T> CreatedTaskWithException<T>(Exception ex)
+        {
+            TaskCompletionSource<T> completion = new TaskCompletionSource<T>();
+            completion.SetException(ex);
+            return completion.Task;
+        }
+
+        static internal Task<T> CreatedTaskWithCancellation<T>()
+        {
+            TaskCompletionSource<T> completion = new TaskCompletionSource<T>();
+            completion.SetCanceled();
+            return completion.Task;
+        }
```

Comment:
Likewise, this can just be:
```C#
return Task.FromCanceled(new CancellationToken(true));
```
If there's a CancellationToken that could be passed in, that could be passed to FromCanceled instead of manufacturing a dummy canceled token.